dev/benchmarks/macrobenchmarks/lib/src/multi_widget_construction.dart

```diff
@@ -33,28 +33,19 @@ class _MultiWidgetConstructTableState extends State<MultiWidgetConstructTable>
   void initState() {
     super.initState();
     controller = AnimationController(
-        vsync: this, duration: const Duration(milliseconds: 10000));
-    curve = CurvedAnimation(parent: controller, curve: Curves.linear)
-      ..addListener(() {
-        final double colorPosition = curve.value;
-        final int c1Position = (colorPosition * (colorList.length + 1)).floor();
-        final Color c1 = colorList[c1Position % colorList.length][900];
-        final Color c2 = colorList[(c1Position + 1) % colorList.length][900];
-        setState(() {
-          baseColor = Color.lerp(
-              c1, c2, colorPosition * (colorList.length + 1) - c1Position);
-        });
-      })
-      ..addStatusListener((AnimationStatus state) {
-        if (state == AnimationStatus.completed) {
-          controller.reverse();
-        } else if (state == AnimationStatus.dismissed) {
-          controller.reset();
-          controller.forward();
-        }
+      vsync: this,
+      duration: const Duration(milliseconds: 10000),
+      lowerBound: 0,
+      upperBound: colorList.length + 1.0,
+    )..addListener(() {
+      final double colorPosition = controller.value;
```

Comment:
The typical way to handle animations is to use an animated builder in your widget tree:

https://api.flutter.dev/flutter/widgets/AnimatedBuilder-class.html

Reply:
Other tests with animation in macrobenchmarks are doing similar thing with `addListener`. I added the modification but let me know if I should keep consistency with other test cases.

Reply:
I see that, and I think I copied their design at some point when I was writing some benchmarks, but those particular pieces of code aren't the suggested approach for our developers to use when creating animations.

We could think about updating the other benchmarks at some point, but the benchmarks aren't meant to be code examples for developers and changing them now could potentially impact their performance and obsolete their long history of accumulated data.

Maybe a compromise would be to add a comment to them indicating that the animations should be managed with AnimatedBuilder widgets, but we are keeping the code as it is to preserve historical measurements.